internal/controller/modelcatalog_controller.go

```diff
@@ -76,6 +79,9 @@ func (r *ModelCatalogReconciler) ensureCatalogResources(ctx context.Context) (ct
 		Namespace: r.TargetNamespace,
 	}
 
+	// Check if we should skip catalog DB creation
+	skipDBCreation := config.GetStringConfigWithDefault(config.SkipModelCatalogDBCreation, "false") == "true"
```

Comment:
This can be moved to main since the env variable won't change on every reconcile.